src/main/java/org/opensearch/flowframework/workflow/HttpHostStep.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+package org.opensearch.flowframework.workflow;
+
+import org.apache.hc.core5.http.HttpHost;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.opensearch.action.support.PlainActionFuture;
+import org.opensearch.core.rest.RestStatus;
+import org.opensearch.flowframework.exception.FlowFrameworkException;
+import org.opensearch.flowframework.util.ParseUtils;
+
+import java.util.Collections;
+import java.util.Locale;
+import java.util.Map;
+import java.util.Set;
+
+import static org.opensearch.flowframework.common.CommonValue.HOSTNAME_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.HTTP_HOST_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.PORT_FIELD;
+import static org.opensearch.flowframework.common.CommonValue.SCHEME_FIELD;
+
+/**
+ * Step to register parameters for an HTTP Connection to a Host
+ */
+public class HttpHostStep implements WorkflowStep {
+
+    private static final Logger logger = LogManager.getLogger(HttpHostStep.class);
+    PlainActionFuture<WorkflowData> hostFuture = PlainActionFuture.newFuture();
+    static final String NAME = "http_host";
+
+    @Override
+    public PlainActionFuture<WorkflowData> execute(
+        String currentNodeId,
+        WorkflowData currentNodeInputs,
+        Map<String, WorkflowData> outputs,
+        Map<String, String> previousNodeInputs
+    ) {
+        Set<String> requiredKeys = Set.of(SCHEME_FIELD, HOSTNAME_FIELD, PORT_FIELD);
+        Set<String> optionalKeys = Collections.emptySet();
```

Comment:
is this just a placeholder for now? I see some other steps might not have this.

Reply:
If credentials are added later, they may be optional.  For now this is just a simple step encoding the Apache HttpHost object.  It takes no other parameters.

Reply:
got it, I am still want to see the full implementation, I was thinking first we handle the steps for the OS endpoints, not sure 100% sure why not to combine all the info together in one place like connector, I am okay merging this for now though

Reply:
I'm still researching the "full implementation" and trying to get known pieces of it done incrementally.   

The "full implementation" will be very similar to how ML Commons does it.

For reference, this workflow step essentially abstracts the HttpHost elements and validation in this class for example: https://github.com/opensearch-project/ml-commons/blob/7b609891a6b0d4df2e1e80959bebdf6f3deb9cdd/ml-algorithms/src/main/java/org/opensearch/ml/engine/httpclient/MLHttpClientFactory.java

Reply:
That sounds good to me, I overall don't see to much of an issue with this, I think though with different examples of how the ratio of httphost steps we will have and the underlying apireststeps it will make more sense. I also as I stated above thought first focus was just the other plugin APIs so less of a need for this but I understand the full vision better